src/general/DayNightCycle.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using Newtonsoft.Json;
+
+[JsonObject(IsReference = true)]
+[UseThriveSerializer]
+public class DayNightCycle
+{
+    [JsonProperty]
+    public DayNightConfiguration LightCycleConfig;
+
+    [JsonProperty]
+    public bool IsEnabled;
+
+    /// <summary>
+    ///   The multiplier used for calculating DayLightPercentage.
+    /// </summary>
+    /// <remarks>
+    ///   This exists as it only needs to be calculated once and
+    ///   the calculation for it is confusing.
+    /// </remarks>
+    [JsonIgnore]
+    private float daytimeMultiplier;
+
+    public DayNightCycle(bool isEnabled)
+    {
+        IsEnabled = isEnabled;
+
+        LightCycleConfig = SimulationParameters.Instance.GetDayNightCycleConfiguration();
+        Time = LightCycleConfig.HoursPerDay / 2;
```

Comment:
So if I get it well, you just instantiate the cycle at noon right?

Reply:
Correct. Starting at night felt weird when starting in the warm pond. Should I add a comment here?

Reply:
Yes please, I think it will help understand it when reading it for the first time.